File: examples/python/src/class/MyStackTest.py
import unittest
from MyStack import MyStack

class MyStackTest(unittest.TestCase):
    def setUp(self):
        self.stack = MyStack(5)

    def test_shouldInsertAndRemoveWithoutProblem(self):
        self.stack.push(100)
        self.assertEquals(self.stack.pop(), 100)
        self.stack.push(200)
        self.stack.push(100)
        self.stack.push(50)
        self.assertEquals(self.stack.pop(), 100)

File: examples/java/mystack/src/test/java/br/edu/ifal/mutationexperiment/mystack/MyStackTest.java
package br.edu.ifal.mutationexperiment.mystack;

//JUnit4
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;


public class MyStackTest  {

  private MyStack stack = new MyStack(5);
    
  @Test  
  public void shouldInsertAndRemoveWithoutProblem() throws Exception {    
    stack.push(100);
    assertEquals(100, stack.pop());
    stack.push(200);
    stack.push(100);
    stack.push(50);
    assertEquals(50, stack.pop());
  }

  @Test  
  public void shouldReturnValueWithoutDelete() throws Exception {
    stack.push(100);
    assertEquals(100, stack.peek());
    assertFalse(stack.isEmpty());
  }

  @Test  
  public void shouldSayStackIsFull() throws Exception {
    stack.push(100);
    stack.push(200);
    stack.push(300);
    stack.push(400);
    stack.push(500);
    assertTrue(stack.isFull());
  }
}